base/src/main/java/com/smartdevicelink/managers/lifecycle/BaseLifecycleManager.java

```diff
@@ -398,6 +399,11 @@ public void onReceived(RPCMessage message) {
                             VehicleType vehicleType = raiResponse.getVehicleType();
                             String systemSoftwareVersion = raiResponse.getSystemSoftwareVersion();
                             if (vehicleType != null || systemSoftwareVersion != null) {
+                                String address = session.getBluetoothMacAddress();
+                                if (address != null && !address.isEmpty()) {
+                                    saveVehicleType(address, vehicleType);
+                                }
```

Comment:
I think we should do it after system info check, am I right?

Reply:
No, it needs to be before, because if vehicle type is not supported, the vehicle type will not be saved to shared preferences, which then on subsequent connections we won't be able to check vehicle type earlier in the RouterServices to potentially pass it off to another RouterService.

Reply:
Thank you for bringing that up, @JulianKast (688d227)